be/src/vec/core/types.h

```diff
@@ -628,141 +676,19 @@ struct Decimal128I : public Decimal<Int128> {
-        value++;
-        return *this;
-    }
-    const Decimal<T>& operator--() {
-        value--;
-        return *this;
-    }
-
-    const Decimal<T>& operator+=(const T& x) {
-        value += x;
-        return *this;
-    }
-    const Decimal<T>& operator-=(const T& x) {
-        value -= x;
-        return *this;
-    }
-    const Decimal<T>& operator*=(const T& x) {
-        value *= x;
-        return *this;
-    }
-    const Decimal<T>& operator/=(const T& x) {
-        value /= x;
-        return *this;
-    }
-    const Decimal<T>& operator%=(const T& x) {
-        value %= x;
-        return *this;
-    }
-
-    static constexpr int max_string_length() { return max_decimal_string_length<T>(); }
-
-    std::string to_string(UInt32 scale) const { return decimal_to_string(value, scale); }
-
-    /**
-     * Got the string representation of a decimal.
-     * @param dst Store the result, should be pre-allocated.
-     * @param scale Decimal's scale.
-     * @param scale_multiplier Decimal's scale multiplier.
-     * @return The length of string.
-     */
-    __attribute__((always_inline)) size_t to_string(char* dst, UInt32 scale,
-                                                    const T& scale_multiplier) const {
-        return decimal_to_string(value, dst, scale, scale_multiplier);
-    }
-
-    T value;
-};
-
 using Decimal32 = Decimal<Int32>;
 using Decimal64 = Decimal<Int64>;
 using Decimal128 = Decimal<Int128>;
 using Decimal256 = Decimal<wide::Int256>;
-template <typename T>
-inline Decimal<T> operator-(const Decimal<T>& x) {
-    return -x.value;
-}
-
-inline Decimal256 operator+(const Decimal256& x, const Decimal256& y) {
-    return Decimal256(x.value + y.value);
-}
-inline Decimal256 operator-(const Decimal256& x, const Decimal256& y) {
-    return Decimal256(x.value - y.value);
-}
-inline Decimal256 operator*(const Decimal256& x, const Decimal256& y) {
-    return Decimal256(x.value * y.value);
-}
-inline Decimal256 operator/(const Decimal256& x, const Decimal256& y) {
-    return Decimal256(x.value / y.value);
-}
-inline Decimal256 operator%(const Decimal256& x, const Decimal256& y) {
-    return Decimal256(x.value % y.value);
-}
-inline Decimal256 operator-(const Decimal256& x) {
-    return Decimal256(-x.value);
-}
 
-inline bool operator<(const Decimal256& x, const Decimal256& y) {
-    return x.value < y.value;
-}
-inline bool operator>(const Decimal256& x, const Decimal256& y) {
-    return x.value > y.value;
-}
-inline bool operator<=(const Decimal256& x, const Decimal256& y) {
-    return x.value <= y.value;
-}
-inline bool operator>=(const Decimal256& x, const Decimal256& y) {
-    return x.value >= y.value;
-}
-inline bool operator==(const Decimal256& x, const Decimal256& y) {
-    return x.value == y.value;
-}
-inline bool operator!=(const Decimal256& x, const Decimal256& y) {
-    return x.value != y.value;
+inline std::strong_ordering operator<=>(const Decimal256& a, const Decimal256& b) {
```

Comment:
not use <=>， the  logic is not very clear.
And it is very dangerous.

Reply:
I think it's more clear than before?